CalPatRec/src/PhiClusterFinder_module.cc

```diff
@@ -0,0 +1,559 @@
+// Function to fill the time spectrum
+//-----------------------------------------------------------------------------
+  void PhiClusterFinder::fillTimeSpectrum(int ClustNo, const std::vector<StrawHitIndex>& ordchcol) {
+    _timespec.Reset();
+    for(unsigned istr=0; istr<ordchcol.size();++istr) {
+      int i = istr;
+      if(_clustno[i]==ClustNo){
+        ComboHit const& ch = (*_chcol)[istr];
+        float time = _t0calc.comboHitTime((*_chcol)[istr],_pitch);
+        _timespec.Fill(time,ch.nStrawHits());
+      }
+    }
+  }
+
+//-----------------------------------------------------------------------------------
+// Function to fill the time spectrum when the hit is not associated to a phi cluster
+//-----------------------------------------------------------------------------------
+  void PhiClusterFinder::fillTimeSpectrumAlt(const std::vector<StrawHitIndex>& ordchcol) {
+    _timespec.Reset();
+    for(unsigned istr=0; istr<ordchcol.size();++istr) {
+      ComboHit const& ch = (*_chcol)[istr];
+      float time = _t0calc.comboHitTime((*_chcol)[istr],_pitch);
+      _timespec.Fill(time,ch.nStrawHits());
+    }
+  }
+
+
+//------------------------------------------------------------------------------
+// Function to find the peak bin the time spectrum
+//-----------------------------------------------------------------------------
+  void PhiClusterFinder::findPeaks(TimeCluster& tc, const std::vector<StrawHitIndex>& ordchcol) {
+    int nbins = _timespec.GetNbinsX()+1;
+    // loop over spectrum to find peaks
+    std::vector<BinContent> bcv;
+    for (int ibin=1;ibin < nbins; ++ibin)
+      if (_timespec.GetBinContent(ibin) >= _ymin) bcv.push_back(make_pair(_timespec.GetBinContent(ibin),ibin));
+    std::sort(bcv.begin(),bcv.end(),[](const BinContent& x, const BinContent& y){return x.first > y.first;});
+    for (const auto& bc : bcv) {
+      float nsh(0.0);
+      float t0(0.0);
+      for (int ibin = std::max(1,bc.second-_npeak);ibin < std::min(nbins,bc.second+_npeak+1); ++ibin) {
+        nsh += _timespec.GetBinContent(ibin);
+        t0 += _timespec.GetBinCenter(ibin)*_timespec.GetBinContent(ibin);
+      }
+      t0 /= nsh;
+      if (nsh >= _minnsh){
+        tc._t0 = TrkT0(t0,_tbin*0.5);
+        tc._nsh = nsh;
+      }
+    }
+    // if(_debug>1) std::cout<<"Find peak T0 =   "<<tc._t0._t0<<" n straw hits = "<<tc._nsh<<std::endl;
+  }
+
+//------------------------------------------------------------------------------
+// Function to assign hits to the time cluster
+//-----------------------------------------------------------------------------
+  void PhiClusterFinder::assignHits(TimeCluster& otc, const std::vector<StrawHitIndex>& ordchcol) {
+    // assign hits to the closest time peak
+    for(unsigned istr=0; istr<ordchcol.size();++istr) {
+      float time = _t0calc.comboHitTime((*_chcol)[istr],_pitch);
+      float dt = fabs(time - otc._t0._t0);
+      if (dt < _maxdt+otc._t0._t0err){
```

Comment:
At this stage, `otc._t0._t0err` is a constant. I think `dt < _maxdt` should be enough.